modules/internal/CString.chpl

```diff
@@ -382,7 +382,8 @@ module CString {
   */
 
   proc c_string.writeThis(x: Writer) {
-    x.write(this);
+    const tmpStr = new string(this, owned=false, needToCopy=false);
+    x.write(tmpStr);
```

Comment:
Now that you mention this, I don't think these are ever going to be called. I'm checking now.